sechub-wrapper-xray/src/main/java/com/mercedesbenz/sechub/wrapper/xray/api/XrayAPIArtifactoryClient.java

```diff
@@ -0,0 +1,116 @@
+        return XrayWrapperArtifactoryClientSupport.ScanStatus.fromString(node.get("status").asText());
+    }
+
+    public boolean requestScanReports() throws XrayWrapperException {
+        XrayAPIRequest request = XrayAPIRequestFactory.createGetScanReportsRequest(xrayWrapperConfiguration.getArtifactory(), artifact);
+        XrayAPIResponse response = send(request);
+        assertNoError(response, "Could not get report from artifactory");
+        return true;
+    }
+
+    public void startArtifactScan() throws XrayWrapperException {
+        XrayAPIRequest request = XrayAPIRequestFactory.createScanArtifactRequest(xrayWrapperConfiguration.getArtifactory(), artifact,
+                xrayWrapperConfiguration.getRegistry());
+        XrayAPIResponse response = send(request);
+        assertNoError(response, "Could not start external Xray scan");
+        JsonNode node = XrayWrapperJSONConverter.get().readJSONFromString(response.getBody());
+        String result = node.get("info").asText();
+        if (!Objects.equals(result, "Scan of artifact is in progress")) {
+            throw new XrayAPIException("Start of scan was not successful: " + result, response.getHttpStatusCode(), response.getResponseMessage(),
+                    response.getBody());
+        }
+    }
+
+    public void deleteArtifact() throws XrayWrapperException {
+        // Xray deletes empty folders with auto cleanup
+        // deletes artifact folder in artifactory
+        XrayAPIRequest request = XrayAPIRequestFactory.createDeleteArtifactRequest(xrayWrapperConfiguration.getArtifactory(), artifact,
+                xrayWrapperConfiguration.getRegistry());
+        XrayAPIResponse response = send(request);
+        assertNoError(response, "Could not delete artifact from repo");
+    }
+
+    public void deleteUploads() throws XrayWrapperException {
+        // deletes _uploads folder in artifactory
+        // the _uploads folder is created when any artifact is uploaded to the
+        // artifactory
+        XrayAPIRequest request = XrayAPIRequestFactory.createDeleteUploadsRequest(xrayWrapperConfiguration.getArtifactory(), artifact,
+                xrayWrapperConfiguration.getRegistry());
+        XrayAPIResponse response = send(request);
+        assertNoError(response, "Could not delete _uploads from artifactory");
+    }
+
+    private XrayAPIResponse send(XrayAPIRequest request) throws XrayWrapperException {
+        HttpURLConnection con = apiHttpUrlConnectionFactory.create(request);
+        return xrayAPIResponseFactory.createHttpResponseFromConnection(con, xrayWrapperConfiguration.getZipDirectory());
+    }
+
+    private boolean isErrorResponse(XrayAPIResponse response) {
+        int statusCode = response.getHttpStatusCode();
+        LOG.debug("Response Code: {}", statusCode);
+        LOG.debug("Response Message: {}", response.getResponseMessage());
+        LOG.debug("Response Headers: {}", response.getHeaders().toString());
```

Comment:
When you call the toString() method in this way, it is always called, even when debug is not enabled. Also it could be possible that headers is null - in this case we had a NPE even for a non-wanted debug message in production.

On the other side: If you give an object to slf4j which is not wellknown, slf4j will always call the `toString` method automatically!

So you can simple change this to:
```suggestion
        LOG.debug("Response Headers: {}", response.getHeaders());
```

The benefit here: toString is only called by slf4j when debugging is enabled. Also null is no problem, because it is handled by slf4j.

#### Just as an information

If you would need something else then `toString` but also only when debugging is enabled, you have to use the `isDebugEnabled` method in this case, to avoid unnecessary object creation. The following code is an example for such handling (but only nonsense output)

```suggestion
        if (LOG.isDebugEnabled()) {
           var headers = response.getHeaders();
           if (headers==null){
                LOG.warn("Headers null, cannot provide debug logging!");
           }else{
                LOG.debug("Response Headers fancy output: {}", headers.createFancyFullDescription());
           }

        }
        LOG.debug("Response Headers: {}", response.getHeaders());
```